typescript/stalin-sort.ts

```diff
@@ -1,10 +1,7 @@
 export function stalinSort(arr: number[]): number[] {
-  let temp = arr[0] - 1;
-  return arr.filter(el => {
-    if (temp < el) {
-      temp = el;
-      return true;
-    }
-    return false;
-  });
-}
+  return arr.reduce((prev: number[], next: number) => {
+    if(prev.length === 0) return [next]
+    if(next >= prev[prev.length - 1]) return [...prev, next]
+    return prev
+  }, []);
```

Comment:
Like your PR  🤟
Can I suggest something like that?
```ts
    return arr.reduce<number[]>((prev, next) =>
        !prev.length || next >= prev[prev.length - 1] ? [...prev, next] : prev
    , [])

Reply:
done